template/src/components/Share.tsx

```diff
@@ -30,76 +30,101 @@ import icons from '../assets/icons';
 import Toast from '../../react-native-toast-message';
 import {BtnTemplate} from '../../agora-rn-uikit';
 import styles from './styles';
+import {useShareLink} from '../pages/ShareLink';
+import {useString} from '../utils/useString';
+import {MeetingInviteInterface} from '../language/default-labels/videoCallScreenLabels';
 
-const Share = (props: any) => {
+const Share = () => {
   const history = useHistory();
-  const {urlView, urlHost, pstn, joinPhrase, roomTitle, hostControlCheckbox} =
-    props;
+  const {
+    attendeeUrl,
+    hostUrl,
+    pstn,
+    joinPhrase,
+    roomTitle,
+    isSeparateHostLink,
+  } = useShareLink((data) => data);
+  const copiedToClipboardText = useString(
+    'copiedToClipboardNotificationLabel',
+  )();
+  const meetingInviteText =
+    useString<MeetingInviteInterface>('meetingInviteText');
+  const meetingUrlText = useString('meetingUrlLabel')();
+  const PSTNNumberText = useString('PSTNNumber')();
+  const PSTNPinText = useString('PSTNPin')();
+  const meetingIdText = useString('meetingIdLabel')();
+  const hostIdText = useString('hostIdLabel')();
   // const {primaryColor} = useContext(ColorContext);
   // const pstn = {number: '+1 206 656 1157', dtmf: '2342'}
   const enterMeeting = () => {
-    if (urlHost) {
+    if (hostUrl) {
       history.push(`/${joinPhrase}`);
     }
   };
 
   const copyToClipboard = () => {
-    Toast.show({text1: 'Copied to Clipboard', visibilityTime: 1000});
-    let stringToCopy = '';
-
-    $config.FRONTEND_ENDPOINT
-      ? hostControlCheckbox
-        ? (stringToCopy += `Meeting - ${roomTitle}
-URL for Attendee: ${$config.FRONTEND_ENDPOINT}/${urlView}
-URL for Host: ${$config.FRONTEND_ENDPOINT}/${urlHost}`)
-        : (stringToCopy += `Meeting - ${roomTitle}
-Meeting URL: ${$config.FRONTEND_ENDPOINT}/${urlHost}`)
-      : platform === 'web'
-      ? hostControlCheckbox
-        ? (stringToCopy += `Meeting - ${roomTitle}
-URL for Attendee: ${window.location.origin}/${urlView}
-URL for Host: ${window.location.origin}/${urlHost}`)
-        : (stringToCopy += `Meeting - ${roomTitle}
-Meeting URL: ${window.location.origin}/${urlHost}`)
-      : hostControlCheckbox
-      ? (stringToCopy += `Meeting - ${roomTitle}
-Attendee Meeting ID: ${urlView}
-Host Meeting ID: ${urlHost}`)
-      : (stringToCopy += `Meeting - ${roomTitle}
-Meeting URL: ${urlHost}`);
-
-    pstn
-      ? (stringToCopy += `PSTN Number: ${pstn.number}
-PSTN Pin: ${pstn.dtmf}`)
-      : '';
+    Toast.show({text1: copiedToClipboardText, visibilityTime: 1000});
```

Comment:
can you break it down further into separate blocks here

Reply:
Done